app/src/main/java/com/ahgitdevelopment/course/customexamples/repository/local/timer/TimerRepositoryImpl.kt

```diff
@@ -0,0 +1,29 @@
+package com.ahgitdevelopment.course.customexamples.repository.local.timer
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.longPreferencesKey
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import javax.inject.Inject
+
+class TimerRepositoryImpl @Inject constructor(
+    private val userPreferencesRepository: DataStore<Preferences>
+) : TimerRepository {
+
+    companion object {
```

Comment:
I normally have this companion object at the bottom of the class

Reply:
nice